apps/gp2-server/src/data-providers/contentful/event.data-provider.ts

```diff
@@ -0,0 +1,346 @@
+        speakerList.push({
+          speaker: undefined,
+          topic: title || undefined,
+        });
+        return speakerList;
+      }
+
+      if (user.onboarded !== true) {
+        return speakerList;
+      }
+
+      speakerList.push({
+        speaker: parseEventSpeakerUser(user),
+        topic: title || undefined,
+      });
+      return speakerList;
+    },
+    [],
+  );
+
+export const parseGraphQLEvent = (
+  item: EventItem,
+): gp2Model.EventDataObject => {
+  if (!item.calendar) {
+    throw new Error(`Event (${item.sys.id}) doesn't have a calendar"`);
+  }
+
+  if (item.status && !isEventStatus(item.status)) {
+    throw new Error(`Invalid event (${item.sys.id}) status "${item.status}"`);
+  }
+
+  const calendar = parseCalendarDataObjectToResponse({
+    ...parseContentfulGraphqlCalendarPartialToDataObject(item.calendar),
+    projects: [],
+    workingGroups: [],
+  });
+
+  const startDate = DateTime.fromISO(item.startDate);
+  const endDate = DateTime.fromISO(item.endDate);
+  const isStale = endDate.diffNow('days').get('days') < -14; // 14 days have passed after the event
+
+  const {
+    sys: { id, publishedAt },
+    title,
+    description,
+    startDateTimeZone,
+    endDateTimeZone,
+    notesPermanentlyUnavailable,
+    notes,
+    videoRecordingPermanentlyUnavailable,
+    videoRecording,
+    presentationPermanentlyUnavailable,
+    presentation,
+    meetingMaterialsPermanentlyUnavailable,
+    meetingMaterials,
+    meetingLink,
+    thumbnail,
+    hideMeetingLink,
+    status,
+    tags,
+    speakersCollection,
+  } = item;
+
+  const speakersItems = (speakersCollection?.items as SpeakerItem[]) ?? [];
+
+  return {
+    id,
+    title: title!,
+    description: description || '',
+    startDate: startDate.toUTC().toString(),
+    startDateTimeZone: startDateTimeZone!,
+    startDateTimestamp: startDate.toSeconds(),
+    endDate: endDate.toUTC().toString(),
+    endDateTimeZone: endDateTimeZone!,
+    endDateTimestamp: endDate.toSeconds(),
+    lastModifiedDate: publishedAt,
+    notes: getContentfulEventMaterial<string, undefined>(
+      notes as RichTextFromQuery,
+      !!notesPermanentlyUnavailable,
+      isStale,
+      undefined,
+    ),
+    videoRecording: getContentfulEventMaterial<string, undefined>(
+      videoRecording as RichTextFromQuery,
+      !!videoRecordingPermanentlyUnavailable,
+      isStale,
+      undefined,
+    ),
+    presentation: getContentfulEventMaterial<string, undefined>(
+      presentation as RichTextFromQuery,
+      !!presentationPermanentlyUnavailable,
+      isStale,
+      undefined,
+    ),
+    meetingMaterials: getContentfulEventMaterial<MeetingMaterial[], []>(
+      meetingMaterials,
+      !!meetingMaterialsPermanentlyUnavailable,
+      isStale,
+      [],
+    ),
+    thumbnail: thumbnail?.url ?? undefined,
+    meetingLink: meetingLink || undefined,
+    hideMeetingLink: hideMeetingLink || false,
+    status: status as EventStatus,
```

Comment:
maybe we don't need the cast as we have done the isEventStatus test earlier.

Reply:
Had to add it otherwise typescript validation complains